SimCalorimetry/HGCalSimProducers/interface/HGCDigitizerBase.h

```diff
@@ -10,157 +10,153 @@
-      updateOutput(coll,rawDataFrame);
-    }
   }
 
   /**
      @short prepares the output according to the number of time samples to produce
-  */
-  void updateOutput(std::auto_ptr<DColl> &coll,D rawDataFrame) {
-    size_t itIdx(4); //index to the in-time digi - this could be configurable in a future version
-
-    //check if in-time sample is above threshold and put result into the event
-    if(doTimeSamples_) {
-      if(rawDataFrame[itIdx].adc() < adcThreshold_ ) return;
-      coll->push_back(rawDataFrame);
-    } else {
-      //create a new data frame containing only the in-time digi
-      D singleRawDataFrame( rawDataFrame.id() );
-      singleRawDataFrame.resize(1);
-
-      HGCSample singleSample;
-      singleSample.set(rawDataFrame[itIdx].gain(),rawDataFrame[itIdx].adc());
-      singleRawDataFrame.setSample(0, singleSample);
-      if(singleRawDataFrame[0].adc() < adcThreshold_ ) return;
-      coll->push_back(singleRawDataFrame);
-    }
-  }
-
-  /**
-     @short applies a shape to each time sample and propagates the tails to the subsequent time samples
    */
-  void runShaper(D &dataFrame) {
-    std::vector<uint16_t> oldADC(dataFrame.size());
-    for(int it=0; it<dataFrame.size(); it++) {
-      uint16_t gain=dataFrame[it].gain();
-      oldADC[it]=dataFrame[it].adc();
-      uint16_t newADC(oldADC[it]);
-	
-      if(shaperN_*shaperTau_>0){
-	for(int jt=0; jt<it; jt++) {
-	  float relTime(bxTime_*(it-jt)+shaperN_*shaperTau_);	
-	  newADC += uint16_t(oldADC[jt]*pow(relTime/(shaperN_*shaperTau_),shaperN_)*exp(-(relTime-shaperN_*shaperTau_)/shaperTau_));	      
-	}
+  void updateOutput(std::auto_ptr<DColl> &coll,D rawDataFrame)
+  {
+    int itIdx(9);
+    if(rawDataFrame.size()<=itIdx+2) return;
+    
+    D dataFrame( rawDataFrame.id() );
+    dataFrame.resize(5);
+    bool putInEvent(false);
+    for(int it=0;it<5; it++) 
+      {
+	HGCSample singleSample;
+	singleSample.set(rawDataFrame[itIdx-2+it].threshold(),
+			 rawDataFrame[itIdx-2+it].mode(),
+			 rawDataFrame[itIdx-2+it].toa(),
+			 rawDataFrame[itIdx-2+it].data());
+	dataFrame.setSample(it, singleSample);
+	if(it==2) { putInEvent=rawDataFrame[itIdx-2+it].threshold(); }
       }
-
-      HGCSample newSample;
-      newSample.set(gain,newADC);
-      dataFrame.setSample(it,newSample);
-    }
+    if(putInEvent) coll->push_back(dataFrame);
   }
 
   /**
      @short to be specialized by top class
    */
-  virtual void runDigitizer(std::auto_ptr<DColl> &coll,HGCSimHitDataAccumulator &simData,uint32_t digitizerType, CLHEP::HepRandomEngine* engine) {
+  virtual void runDigitizer(std::auto_ptr<DColl> &coll,HGCSimHitDataAccumulator &simData,uint32_t digitizerType, CLHEP::HepRandomEngine* engine)
+  {
     throw cms::Exception("HGCDigitizerBaseException") << " Failed to find specialization of runDigitizer";
   }
 
   /**
      @short DTOR
-  */
-  ~HGCDigitizerBase() { };
+   */
+  ~HGCDigitizerBase() 
+    { };
   
   //baseline configuration
```

Comment:
(not reco code, but the comment is general)
the data members appear to be public, which violates our programming rules